pkg/cmd/container/logs.go

```diff
@@ -14,93 +14,42 @@
    limitations under the License.
 */
 
-package main
+package container
 
 import (
 	"context"
 	"fmt"
 	"os"
 	"os/signal"
-	"strconv"
 	"syscall"
 
 	"github.com/containerd/containerd"
+	"github.com/containerd/nerdctl/pkg/api/types"
 	"github.com/containerd/nerdctl/pkg/api/types/cri"
 	"github.com/containerd/nerdctl/pkg/clientutil"
 	"github.com/containerd/nerdctl/pkg/idutil/containerwalker"
 	"github.com/containerd/nerdctl/pkg/labels"
 	"github.com/containerd/nerdctl/pkg/labels/k8slabels"
 	"github.com/containerd/nerdctl/pkg/logging"
 	"github.com/sirupsen/logrus"
-	"github.com/spf13/cobra"
 )
 
-func newLogsCommand() *cobra.Command {
-	var logsCommand = &cobra.Command{
-		Use:               "logs [flags] CONTAINER",
-		Args:              IsExactArgs(1),
-		Short:             "Fetch the logs of a container. Currently, only containers created with `nerdctl run -d` are supported.",
-		RunE:              logsAction,
-		ValidArgsFunction: logsShellComplete,
-		SilenceUsage:      true,
-		SilenceErrors:     true,
-	}
-	logsCommand.Flags().BoolP("follow", "f", false, "Follow log output")
-	logsCommand.Flags().BoolP("timestamps", "t", false, "Show timestamps")
-	logsCommand.Flags().StringP("tail", "n", "all", "Number of lines to show from the end of the logs")
-	logsCommand.Flags().String("since", "", "Show logs since timestamp (e.g. 2013-01-02T13:23:37Z) or relative (e.g. 42m for 42 minutes)")
-	logsCommand.Flags().String("until", "", "Show logs before a timestamp (e.g. 2013-01-02T13:23:37Z) or relative (e.g. 42m for 42 minutes)")
-	return logsCommand
-}
-
-func logsAction(cmd *cobra.Command, args []string) error {
-	globalOptions, err := processRootCmdFlags(cmd)
-	if err != nil {
-		return err
-	}
-	dataStore, err := clientutil.DataStore(globalOptions.DataRoot, globalOptions.Address)
+func Logs(ctx context.Context, container string, options types.ContainerLogsOptions) error {
```

Comment:
Can you add a brief comment here? 

nit: also maybe change `container` name to something else (for example, `c`, `arg`), since the package name is also `container`. Not sure if there will any unexpected conflict.

Reply:
> Can you add a brief comment here?

Do you mean adding a function comment for `Logs`? If yes, maybe we'd want to do it consistently. Currently it seems that no other function has such comment.

> also maybe change container name to something else (for example, c, arg)

Maybe `ctr`? I feel that `arg` may be too generic, and `c` is a bit too short for a parameter.

Reply:
`ctr` also means `ctr` cli from containerd, which might be confusing. I'm okay with keeping `container` if there is no conflict.

Reply:
Rebased.

> I'm okay with keeping container if there is no conflict.

Keeping it as is as there's no conflict.